readme.md

```diff
@@ -2,7 +2,7 @@
 
 > [Memoize](https://en.wikipedia.org/wiki/Memoization) functions - An optimization used to speed up consecutive function calls by caching the result of calls with identical input
 
-Memory is automatically released when an item expires.
+The input is cached by value (if supported by `JSON.stringify`) unless it’s a single argument of type `function`, in which case it’s stored by reference. Memory is automatically released when an item expires.
```

Comment:
I think it should be documented here instead: https://github.com/sindresorhus/mem#cachekey

Reply:
Indeed, I noticed that there's plenty of overlap though, to the point that the comment about functions is the only missing part, and even then it's implied by "used directly as key"